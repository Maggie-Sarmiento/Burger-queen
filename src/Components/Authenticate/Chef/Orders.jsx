/* eslint-disable react/prop-types */
import { Container, Grid, Typography } from '@mui/material';
import React, { useEffect, useState } from 'react';
import CardOrder from './CardOrder';
import style from '../Admin/Menu/Menu.module.css';

const Orders = () => {
  const [dataMenu, setDataMenu] = useState([]);
  console.log('orders data menu', dataMenu);

  useEffect(() => {
    fetch('http://localhost:3001/orders')
      .then((response) => response.json())
      .then((data) => setDataMenu(data));
  }, []);

  // eslint-disable-next-line prefer-destructuring
  const products = dataMenu.products;
  console.log('orders products', products);

  return (
    <section className={style.sectionMenu}>
      <Container>
        <>
          <Typography
            variant="h4"
            component="h2"
            marginBottom={3}
            marginTop={5}
          >
            Pedidos
          </Typography>
          <Grid container spacing={3}>
            {dataMenu.map((order) => (
              <CardOrder
                key={order.id}
                // table={table}
                order={order}
              />
            ))}
          </Grid>
        </>
      </Container>
    </section>
  );
};

export default Orders;
