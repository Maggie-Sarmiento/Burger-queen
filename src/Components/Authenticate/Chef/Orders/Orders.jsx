/* eslint-disable react/prop-types */
import { Container, Grid, Typography } from '@mui/material';
import React, { useEffect, useState } from 'react';
import CardOrder from '../CardOrder/CardOrder';
import style from '../../Admin/Menu/Menu.module.css';

const Orders = () => {
  const [dataMenu, setDataMenu] = useState([]);
  const [refreshData, setRefreshData] = useState(false);
  console.log('orders data menu', dataMenu);

  useEffect(() => {
    fetch('http://localhost:3001/orders')
      .then((response) => response.json())
      .then((data) => setDataMenu(data));
  }, [refreshData]);

  // eslint-disable-next-line prefer-destructuring
  const products = dataMenu.products;
  console.log('orders products', products);

  return (
    <section className={style.sectionMenu}>
      <Container>
        <>
          <Typography
            variant="h4"
            component="h2"
            marginBottom={3}
            marginTop={5}
          >
            Pedidos
          </Typography>
          <Grid container spacing={3}>
            {dataMenu.filter((elem) => elem.status === 'process').map((order) => (
              <CardOrder
                key={order.id}
                // table={table}
                order={order}
                id={order.id}
                refreshData={refreshData}
                setRefreshData={setRefreshData}
              />
            ))}
          </Grid>
        </>
      </Container>
    </section>
  );
};

export default Orders;
