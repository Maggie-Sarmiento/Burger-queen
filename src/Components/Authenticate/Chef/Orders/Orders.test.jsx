/* eslint-disable no-undef */
import React from 'react';
import { act, create } from 'react-test-renderer';
import Orders from './Orders';

describe('test Orders', () => {
  test('render Orders', () => {
    let root;
    act(() => {
      root = create(
        <Orders />,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
