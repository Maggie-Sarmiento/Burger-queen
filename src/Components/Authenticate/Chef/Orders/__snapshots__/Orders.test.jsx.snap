// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Orders render Orders 1`] = `
<section
  className="sectionMenu"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h4 css-10mgom-MuiTypography-root"
    >
      Pedidos
    </h2>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    />
  </div>
</section>
`;
